test/lib/bamboo/adapters/test_adapter_test.exs

```diff
@@ -141,6 +141,13 @@ defmodule Bamboo.TestAdapterTest do
     end
   end
 
+  test "assert_delivered_with allows regex matching" do
+    new_email(to: {nil, "foo@bar.com"}, from: {nil, "foo@bar.com"}, text_body: "I really like coffee")
+      |> TestMailer.deliver_now
+
+    assert_delivered_with text_body: ~r/like/
```

Comment:
Could you also add something similar to https://github.com/thoughtbot/bamboo/pull/260/files#diff-162d78c24e2d8c7dfed124276ea1f3d6L114 that tests for emails that'd don't have a matching body? That way the tests are a bit more reliable